dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java

```diff
@@ -2178,16 +2178,15 @@ protected boolean filterAdd(HeaderKeys key, byte[] value) {
     private boolean isPrivateHeaderTrusted(HeaderKeys key) {
         HttpServiceContextImpl hisc = getServiceContext();
         InetAddress remoteAddr = null;
-        String address = null;
-        if (hisc != null && (remoteAddr = hisc.getRemoteAddr()) != null) {
-            address = remoteAddr.getHostAddress();
+        if (hisc != null) {
+            remoteAddr = hisc.getRemoteAddr();
         }
 
-        boolean trusted = HttpDispatcher.usePrivateHeaders(address, key.getName());
+        boolean trusted = HttpDispatcher.usePrivateHeaders(remoteAddr, key.getName());
 
         if (!trusted) {
             if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-                Tr.debug(tc, "isPrivateHeaderTrusted: " + key.getName() + " is not trusted for host " + address);
+                Tr.debug(tc, "isPrivateHeaderTrusted: " + key.getName() + " is not trusted for host " + remoteAddr.getHostAddress());
```

Comment:
Will 'trusted' be true if remoteAddr==null? Otherwise this line might potentially NPE since remoteAddr may not have been initialized in lines 2181-2182.